app/javascript/packs/admin/editUserActions.jsx

```diff
@@ -0,0 +1,42 @@
+import { AdminUserActionsModal } from '@admin/AdminUserActionsModal';
+import { initializeDropdown } from '@utilities/dropdownUtils';
+
+let preact;
+let Modal;
+
+const userEditActionsMenu = document.getElementById('options-dropdown');
+
+initializeDropdown({
+  triggerElementId: 'options-dropdown-trigger',
+  dropdownContentId: 'options-dropdown',
+});
+
+userEditActionsMenu.addEventListener('click', async (event) => {
+  const modalContainer = document.createElement('div');
+  document.body.appendChild(modalContainer);
+
+  const { target: potentialButton } = event;
+  if (
+    potentialButton.tagName !== 'BUTTON' ||
+    Object.prototype.hasOwnProperty.call(potentialButton.dataset, 'noModal')
+  ) {
+    // The button that was clicked does not require a modal.
+    return;
+  }
+
+  // Only load Preact if we haven't already.
+  if (!preact || !AdminUserActionsModal) {
```

Comment:
As the comment states, we only load Preact and the component, when we need it. I put this in the click handler because an admin may never click on a modal. When they do, there will be a small hit to dynamically import Preact and the modal. This only happens once per lifecycle of the page.